pkg/database/cleanup.go

```diff
@@ -107,7 +107,7 @@ type cleanupWhere struct {
 }
 
 func (db *Database) PeriodicCleanup(ctx context.Context, stmt CleanupStmt) error {
-	g, ctxCleanup := errgroup.WithContext(ctx)
+	g, _ := errgroup.WithContext(ctx)
```

Comment:
Instead of unnecessarily creating an errgroup, we could simply do the following at the end of the function:

```
select {
    case err := <-errs:
        return err
    case <-ctx.Done():
        return ctx.Err()
}
```